src/Repository/DefaultUnleashRepository.php

```diff
@@ -51,22 +53,38 @@ public function findFeature(string $featureName): ?Feature
     public function getFeatures(): iterable
     {
         if (!$features = $this->getCachedFeatures()) {
-            $request = $this->requestFactory
-                ->createRequest('GET', $this->configuration->getUrl() . 'client/features')
-                ->withHeader('UNLEASH-APPNAME', $this->configuration->getAppName())
-                ->withHeader('UNLEASH-INSTANCEID', $this->configuration->getInstanceId());
+            if (!$this->configuration->isFetchingEnabled()) {
+                if (!$data = $this->getBootstrappedResponse()) {
+                    throw new LogicException('Fetching of Unleash api is disabled but no bootstrap is provided');
+                }
+            } else {
+                $request = $this->requestFactory
+                    ->createRequest('GET', $this->configuration->getUrl() . 'client/features')
+                    ->withHeader('UNLEASH-APPNAME', $this->configuration->getAppName())
+                    ->withHeader('UNLEASH-INSTANCEID', $this->configuration->getInstanceId());
+
+                foreach ($this->configuration->getHeaders() as $name => $value) {
+                    $request = $request->withHeader($name, $value);
+                }
 
-            foreach ($this->configuration->getHeaders() as $name => $value) {
-                $request = $request->withHeader($name, $value);
+                try {
+                    $response = $this->httpClient->sendRequest($request);
+                    if ($response->getStatusCode() === 200) {
+                        $data = $response->getBody()->getContents();
+                    }
+                } catch (Exception $exception) {
+                    // empty catch, $data does not exist and will be handled below
+                }
+                $data ??= $this->getBootstrappedResponse();
```

Comment:
Nice, I have not seen that operator before! 

Would it make sense to validate if it is anything more than an empty string? 
What if it is an empty list of feature toggles?

Reply:
Empty string couldn't pass here, an exception would be thrown somewhere in the providers. Empty feature toggle list is a valid response, no? For example, if your unleash server goes down you want to play it safe and don't let anyone see a list of any flag guarded features.

I can add some validator that it at least is in valid format to fail early, but I think we shouldn't check for more than the data having a "features" key and it being an array.

Reply:
> but I think we shouldn't check for more than the data having a "features" key and it being an array.

Agreed!

Reply:
@ivarconr I added the check in the latest commit.